src/devices/Seesaw/SeesawGpio.cs

```diff
@@ -0,0 +1,150 @@
+            if (!HasModule(Seesaw_Module.Gpio))
+            {
+                throw new NotSupportedException($"The hardware on I2C Bus {I2cDevice.ConnectionSettings.BusId}, Address 0x{I2cDevice.ConnectionSettings.DeviceAddress:X2} does not support Adafruit SeeSaw GPIO functionality");
+            }
+
+            byte[] buffer = new byte[] { (byte)(pins >> 24), (byte)(pins >> 16), (byte)(pins >> 8), (byte)pins, (byte)(pins >> 56), (byte)(pins >> 48), (byte)(pins >> 40), (byte)(pins >> 32)};
+
+            switch (mode)
+            {
+                case PinMode.Output:
+                    Write(Seesaw_Module.Gpio, Seesaw_Function.Gpio_Dirset_Bulk, buffer);
+                    break;
+
+                case PinMode.Input:
+                    Write(Seesaw_Module.Gpio, Seesaw_Function.Gpio_Dirclr_Bulk, buffer);
+                    break;
+
+                case PinMode.InputPullUp:
+                    Write(Seesaw_Module.Gpio, Seesaw_Function.Gpio_Dirclr_Bulk, buffer);
+                    Write(Seesaw_Module.Gpio, Seesaw_Function.Gpio_Pullenset, buffer);
+                    Write(Seesaw_Module.Gpio, Seesaw_Function.Gpio_Bulk_Set, buffer);
+
+                    break;
+
+                case PinMode.InputPullDown:
+                    Write(Seesaw_Module.Gpio, Seesaw_Function.Gpio_Dirclr_Bulk, buffer);
+                    Write(Seesaw_Module.Gpio, Seesaw_Function.Gpio_Pullenset, buffer);
+                    Write(Seesaw_Module.Gpio, Seesaw_Function.Gpio_Bulk_Clr, buffer);
+                    break;
+            }
+
+        }
+
+        /// <summary>
+        /// Write a value to GPIO pin
+        /// </summary>
+        /// <param name="pin">The pin that has its value set.</param>
+        /// <param name="value">The pin value to be set.</param>
+        public void WriteGpioDigital(byte pin, bool value)
+        {
+            if (pin < 0 || pin > 63)
+            {
+                throw new ArgumentException("Gpio pin must be within 0-63 range.");
+            }
+
+            WriteGpioDigitalBulk((UInt64)(1 << pin), value);
+        }
+
+        /// <summary>
+        /// Write a value to a number of GPIO pins
+        /// </summary>
+        /// <param name="pins">A 64bit integer containing 1 bit for each pin. If a bit is set to 1 then the pin value is set for the associated pin.</param>
+        /// <param name="value">The pin value to be set.</param>
+        public void WriteGpioDigitalBulk(UInt64 pins, bool value)
+        {
+            if (!HasModule(Seesaw_Module.Gpio))
+            {
+                throw new NotSupportedException($"The hardware on I2C Bus {I2cDevice.ConnectionSettings.BusId}, Address 0x{I2cDevice.ConnectionSettings.DeviceAddress:X2} does not support Adafruit SeeSaw GPIO functionality");
+            }
+
+            byte[] buffer = new byte[] { (byte)(pins >> 24), (byte)(pins >> 16), (byte)(pins >> 8), (byte)pins, (byte)(pins >> 56), (byte)(pins >> 48), (byte)(pins >> 40), (byte)(pins >> 32) };
```

Comment:
https://docs.microsoft.com/en-us/dotnet/api/system.buffers.binary.binaryprimitives?view=netstandard-2.1

Reply:
@krwq Unfortunately the byte array is not formatted as a ulong converted ty bytes but as two uints representing the low and high 32 bits and so there is no obvious bit conversion that I can see.
I have created functions for this conversion and documented the byte array layout and it will be available in the next update